class/practice_3.1/Helpers.cpp

```diff
@@ -0,0 +1,29 @@
+#include "Helpers.hpp"
+#include <algorithm>
+#include <string>
+#include <iostream>
+#include <vector>
+#include <ranges>
+
+double dummyFunc(double x) {
+    return x > 0 ? x : 0;
+}
+
+std::string repeatString(const std::vector<int>& numbers, const std::string& str) {
+    if (numbers.empty()) {
+        return "";
+    }
+    
+    auto repeatedView = str | std::views::repeat;
+
+    auto concatenatedView = numbers | std::views::transform(&repeatedView {
+        return repeatedView | std::views::take(num);
+    }) | std::views::join;
```

Comment:
just add | std::ranges::to<std::string> and it will give you the expected result.
In general code can be something like
```
    auto repeatCount = numbers.empty() ? 0 : std::ranges::max(numbers);
    return repeatCount > 0
            ? std::views::repeat(str, repeatCount) | std::views::join | std::ranges::to<std::string>()
            : "";
```